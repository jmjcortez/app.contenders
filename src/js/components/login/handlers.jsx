import axios from 'axios';

import { 
    userAuthenticateAction, 
    userAuthenticateSuccessAction,
    userAuthenticateFailAction,
} from '../../actions/user-actions';

export const API = 'http://localhost:8000';

export const handleAuthenticate = async (values, dispatch) => {
    dispatch(userAuthenticateAction());

    try {
        const response = await axios.post(`${API}/auth/token/`, values);
        
        dispatch(userAuthenticateSuccessAction(response.data));
        
    } catch (error) {
        dispatch(userAuthenticateFailAction(error));
    }
};
