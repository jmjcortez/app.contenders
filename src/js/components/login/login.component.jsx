import React from "react";
import { Field, reduxForm } from "redux-form";
import { Redirect } from 'react-router-dom';
import { Form, Button } from 'react-bootstrap';
import styled from 'styled-components';

import { FieldInput } from '../form-fields';

const Container = styled.div`
    margin-top: 25px;
`;

const FormWrapper = styled.div`
    max-width: 535px;
    padding: 64px 48px;
    background-color: ${props => props.theme.colors.pink}
    margin: auto;
`;

const StyledTitle = styled.h1`
    font-size: 48px;
    font-family: 'Roboto Condensed', sans-serif;
    font-weight: 700;
    line-height: 1;
    text-transform: uppercase;
    color: ${props => props.theme.colors.black} !important;
    text-align: center;
`;

const Underline = styled.span`
    width: 73px;
    height: 4px;
    margin: 8px auto 0;
    display: block;
    background-color: ${props => props.theme.colors.light};
`;

const Subtitle = styled.div`
    font-size: 14px;
    font-family: ${props => props.theme.fonts.roboto.fontFamily};
    font-weight: 300;
    line-height: 1.33;
    text-align: center;
    color: ${props => props.theme.colors.black} !important;
    margin-bottom: 48px;
    margin-top: 14px;
`;


const Link = styled.a`
    color: ${props => props.theme.colors.black} !important;
    text-decoration: underline;
`;

const FormLabel = styled(Form.Label)`
    font-family: ${props => props.theme.fonts.roboto.fontFamily};
`;

const StyledButton = styled(Button)`
    margin-top: 30px;
    background-color: ${props => props.theme.colors.light} !important;
    border-color: ${props => props.theme.colors.light} !important;
`;

export const Login = props => {
    const { handleSubmit } = props
    
    if (window.localStorage.getItem("JWT")) {
        return <Redirect to={{pathname: "/homepage"}} />
    };

    return (
        <Container>
            <FormWrapper>
                <StyledTitle>LOG IN</StyledTitle>
                <Underline />
                <Subtitle>
                    Not a member yet? <Link href='register'>Sign Up here</Link>
                </Subtitle>
                <Form onSubmit={handleSubmit}>
                    <Form.Group controlId="formBasicEmail">
                        <FormLabel>Email</FormLabel>  
                        <Field 
                            name='email'
                            type='email' 
                            component={FieldInput}
                        />
                        <Form.Text className="text-muted">
                        Required
                        </Form.Text>
                    </Form.Group>

                    <Form.Group controlId="formBasicPassword">
                        <Form.Label>Password</Form.Label>
                        <Field 
                            name='password'
                            type='password' 
                            component={FieldInput}
                        />
                    </Form.Group>
                        <Form.Text className="text-muted">
                            Required
                        </Form.Text>
                    <StyledButton variant="primary" type="submit" block>
                        Submit
                    </StyledButton>
                </Form>
            </FormWrapper>
        </Container>
    )

};

export default reduxForm({
    form: 'login' // a unique identifier for this form
})(Login);