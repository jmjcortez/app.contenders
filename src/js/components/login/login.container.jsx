import { connect } from "react-redux";
import { compose, withHandlers } from "recompose";

import Login from "./login.component";
import { userAuthenticateAction } from "../../actions/user-actions";

const mapStateToProps = (state) => {
    return {
        aklas: 'bading ka',
    }
};

// const handlers = withHandlers({
//     authenticateUser: ({ dispatch }) => () => dispatch(userAuthenticateAction()),
// });

const mapDispatchToProps = dispatch => {
  return {
    // dispatching actions returned by action creators
    onSubmit: values => dispatch(userAuthenticateAction(values)),
  }
}

export default compose(
    connect(mapStateToProps, mapDispatchToProps),
    // handlers
)(Login);

