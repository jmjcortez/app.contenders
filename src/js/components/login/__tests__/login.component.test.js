import React from 'react';
import { shallow } from 'enzyme';
import { Redirect } from 'react-router-dom';

import { Login } from '../login.component';

describe('Login component test', () => {
    afterEach(() => {
        window.localStorage.clear();
    });

    it('should render correctly with no props', () => {
        const component = shallow(<Login />);
        
        expect(component).toMatchSnapshot();
    });
});