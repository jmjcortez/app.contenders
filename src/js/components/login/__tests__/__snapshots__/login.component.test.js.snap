// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field input test should render correctly with no props with input prop 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="login"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;

exports[`Login component test should render correctly with no props 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="login"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;

exports[`Login component test should render correctly with no props with input prop 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="login"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
